pkg/feature/templates/servicemesh/authorino/mesh-authz-ext-provider.patch.tmpl

```diff
@@ -0,0 +1,13 @@
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: {{ .ControlPlane.Name }}
+  namespace: {{ .ControlPlane.Namespace }}
+spec:
+  techPreview:
+    meshConfig:
+      extensionProviders:
+      - name: {{ .AppNamespace }}-odh-auth-provider
```

Comment:
I think we can simplify names here. See my patch ^

Reply:
So, I applied your patch in commit 773cef9.
The only thing that was a small surprise for me was updating in-place the DSCI fields. So, I though the preference was to leave the source (the DSCI or DSC) unchanged and use fields of the feature, since updating in-place may have an impact about what values other `Features` would see (and also what the controller would see once the features run).

That said, I'm not opposed on doing it. So, I simply applied the patch. Code is less confusing without the replicated fields.